courses/serializers.py

```diff
@@ -149,3 +150,47 @@ class Meta:
             'requirements',
             'sections',
         ]
+
+
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = [
+            'email',
+            'phone'
+        ]
+
+
+class UserSerializer(serializers.ModelSerializer):
+    profile = UserProfileSerializer(read_only=False)
+
+    def update(self, instance, validated_data):
+        prof, _ = UserProfile.objects.get_or_create(user=instance)
+        prof_data = validated_data.pop('profile')
+        for key in ['username', 'first_name', 'last_name']:
+            if key in validated_data:
+                setattr(instance, key, validated_data[key])
+        for key in ['phone', 'email']:
+            if key in prof_data:
+                setattr(prof, key, prof_data[key])
+        prof.save()
+        setattr(instance, 'profile', prof)
+        instance.save()
+        return instance
+
+    class Meta:
+        model = get_user_model()
+        fields = [
+            'username',
+            'first_name',
+            'last_name',
+            'profile'
+        ]
+
+
+class StatusUpdateSerializer(serializers.ModelSerializer):
+    section = serializers.ReadOnlyField(source='section__full_code')
```

Comment:
Just a reminder to check and make sure we're not issuing one DB query per StatusUpdate, since we're accessing a related field here. You can check this in Django Debug Toolbar, slack me if u can't find it